File: leet_BST_0129_sum.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */

func sumNumbers(root *TreeNode) int {
    return helper(root, 0)
}

func helper(node *TreeNode, curr int) int {
    if node == nil {
        return 0
    }
    curr = curr * 10 + node.Val
    if node.Left == nil && node.Right == nil {
        return curr
    }
    return helper(node.Left, curr) + helper(node.Right, curr)
}

File: leet_BST_0129_sum.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */

var sumNumbers = function(root) {
    var helper = function(node, curr) {
        if (node == null)
            return 0
        let res = curr * 10 + node.val
        if (node.left == null && node.right == null)
            return res
        return helper(node.left, res) + helper(node.right, res)
    }
    return helper(root, 0)
};
